R/setThreadCount.r

```diff
@@ -19,44 +21,45 @@
 #' # Finding the best performing thread count for each benchmarked data.table function with a data size of 10000000 rows and 10 columns:
 #' benchmarkData <- data.table.threads::findOptimalThreadCount(1e7, 10)
 #' # Setting the optimal thread count for the 'forder' function:
-#' setThreadCount(benchmarkData, "forder", "optimal")
-#' # Can verify by checking benchmarkData and getDTthreads()
+#' setThreadCount(benchmarkData, "forder", efficiencyFactor = 0)
+#' getDTthreads()
 #' # Setting the recommended thread count for the 'nafill' function:
-#' setThreadCount(benchmarkData, "nafill", "recommended")
+#' setThreadCount(benchmarkData, "nafill", efficiencyFactor = 0.5)
+#' getDTthreads()
 #' }
 
-setThreadCount <- function(benchmarkData, functionName, type = "recommended", verbose = FALSE)
+setThreadCount <- function(benchmarkData, functionName, efficiencyFactor = 0.5, verbose = FALSE)
 {
+  if(!is.numeric(efficiencyFactor) || efficiencyFactor < 0 || efficiencyFactor > 1) {
+    stop("Invalid efficiencyFactor specified. (Please use 0.5 for recommended thread count or 0 for optimal thread count!)")
+  }
+
   setDTthreads(
-    if(type == "optimal")
+    if(efficiencyFactor == 0)
     {
       fastestMedianTime <- benchmarkData[expr == functionName, .(median = min(median))]
       bestThreadCount <- benchmarkData[expr == functionName & median == fastestMedianTime$median, threadCount]
-      if(verbose) 
+      if(verbose)
       {
-        message(sprintf("The number of threads that data.table will use has been set to %d, the thread count that achieved the best runtime for data.table::%s() based on the performed benchmarks.\n", bestThreadCount, functionName))
+        message("The number of threads that data.table will use has been set to ", bestThreadCount, ", the thread count that achieved the best runtime for data.table::", functionName, "() based on the performed benchmarks.")
       }
       bestThreadCount
     }
-    else if(type == "recommended")
+    else if(efficiencyFactor == 0.5)
```

Comment:
why do you need if efficient == 0.5 here ?
shouldn't it work for 0.25 or etc?

Reply:
Yup it should work for other values too, but I was confused on whether to have it as a replacement for just the two types (optimal/recommended or 0/0.5 value) or in general for any value within [0, 1] (which is why I kept that condition and asked you above)

Reply:
great then please remove the if else and just use else.
and also can you handle optimal as efficiency=0 instead of a special case?
if so that would make the code easier to understand and maintain